src/CattyTests/PenFeatureTests.swift

```diff
@@ -0,0 +1,254 @@
+
+        scene.update(CACurrentMediaTime())
+        scene.update(CACurrentMediaTime())
+        scene.update(CACurrentMediaTime())
+
+        scene.update(CACurrentMediaTime())
+        scene.update(CACurrentMediaTime())
+
+        XCTAssertEqual(scene.children.count, 1)
+
+        scene.stopProject()
+
+    }
+
+    func testWhenSpritePositionNotChangedPenDown() {
+
+        let scene = SceneBuilder(project: self.project).build()
+        let sceneStarted = scene.startProject()
+        XCTAssertEqual(sceneStarted, true)
+
+        guard let spriteNode = scene.childNode(withName: "testName") as? CBSpriteNode else {
+            return
+        }
+
+        spriteNode.penConfiguration.penDown = true
+        spriteNode.catrobatPosition = initialPosition
+
+        scene.update(CACurrentMediaTime())
+        scene.update(CACurrentMediaTime())
+        scene.update(CACurrentMediaTime())
+
+        scene.update(CACurrentMediaTime())
+        scene.update(CACurrentMediaTime())
+
+        XCTAssertEqual(scene.children.count, 1)
+
+        scene.stopProject()
+
+    }
+
+    func testDrawnLineAttributes() {
+
+        let scene = SceneBuilder(project: self.project).build()
+        XCTAssertEqual(scene.startProject(), true)
+
+        guard let spriteNode = scene.childNode(withName: "testName") as? CBSpriteNode else {
+            XCTAssertEqual(1, 0) // Test fails when spriteNode is nil
+            return
+        }
+
+        spriteNode.penConfiguration.penDown = true
+        spriteNode.penConfiguration.color = UIColor(red: 0.0, green: 0.0, blue: 0.0, alpha: 1.0)
+        spriteNode.penConfiguration.size = 5.0
+        spriteNode.catrobatPosition = initialPosition
+
+        scene.update(CACurrentMediaTime())
+        scene.update(CACurrentMediaTime())
+        scene.update(CACurrentMediaTime())
+
+        spriteNode.catrobatPosition = CGPoint(x: 1, y: 1)
+
+        scene.update(CACurrentMediaTime())
+        scene.update(CACurrentMediaTime())
+
+        XCTAssertEqual(scene.children.count, 2)
+
+        guard let line = scene.childNode(withName: SpriteKitDefines.penShapeNodeName) as? SKShapeNode else {
+            XCTAssertEqual(1, 0)
+            return
+        }
+
+        XCTAssertEqual(line.lineWidth, 5.0)
+        XCTAssertEqual(line.strokeColor, UIColor(red: 0.0, green: 0.0, blue: 0.0, alpha: 1.0))
+
+        scene.stopProject()
+
+    }
+
+    func testPenFeatureWithMotionBrick() {
+
+        let scene = SceneBuilder(project: self.project).build()
+        XCTAssertEqual(scene.startProject(), true)
+
+        guard let spriteNode = scene.childNode(withName: "testName") as? CBSpriteNode else {
+            XCTAssertEqual(1, 0) // Test fails when spriteNode is nil
+            return
+        }
+
+        spriteNode.penConfiguration.penDown = true
+        spriteNode.catrobatPosition = CGPoint.zero
+
+        let script = WhenScript()
+        script.object = spriteObject
+
+        let brick = SetXBrick()
```

Comment:
Please try to use as little "dependencies" in your test as possible. The test should not fail whenever the `SetXBrick` or the `WhenScript` have a problem. The test should be independent of these autonomous Bricks. Luckily, this functionality can be tested independently which would make the test more readable and also less error-prone. My suggestion would be update the SpriteNode's position directly (`spriteNode.position = `).